thrust/thrust/functional.h

```diff
@@ -891,14 +892,30 @@ struct identity
    */
   using result_type = T;
 
-  /*! Function call operator. The return value is <tt>x</tt>.
-   */
   _CCCL_EXEC_CHECK_DISABLE
   _CCCL_HOST_DEVICE constexpr const T& operator()(const T& x) const
   {
     return x;
   }
-}; // end identity
+
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_HOST_DEVICE constexpr T& operator()(T& x) const
+  {
+    return x;
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_HOST_DEVICE constexpr const T&& operator()(const T&& x) const
+  {
+    return x;
```

Comment:
This should actually move
```suggestion
    return _CUDA_VSTD::move(x);
```